src/ExpressReceiver.ts

```diff
@@ -25,21 +25,23 @@ export interface ExpressReceiverOptions {
   processBeforeResponse?: boolean;
   clientId?: string;
   clientSecret?: string;
-  stateSecret?: string; // ClearStateStoreOptions['secret']; // required when using default stateStore
-  installationStore?: InstallationStore; // default MemoryInstallationStore
-  scopes?: string | string[];
+  stateSecret?: InstallProviderOptions['stateSecret']; // ClearStateStoreOptions['secret']; // required when using default stateStore
```

Comment:
i'm a little torn about the choice to reference types from inside `@slack/oauth` to define these property types.

on the one hand, it clearly represents the relationship of these properties to the underlying library. that's important information that otherwise isn't clear.

on the other hand, two things: first, it's not as nice for a developer who is introspecting types. in order to find the type of `InstallProviderOptions['stateSecret'']`, i need to leave this package and go look in another package, just to learn its a simple `string`. and second, its hazardous to directly export the interface of another package since it can change independently of this package. for example, if tomorrow `InstallURLOptions['scopes']` changes in a backwards incompatible way, and we kept the type `string | string[]` in here, our build would break to tell us this doesn't work, and we could write some code to transform the input to the type the underlying package accepts. But if we're just reference the type, there's no failure. So (re)defining our package's types acts as a "facade" so we can control the interface, and possibly write some code here to adapt for changes in the underlying library.

what do you think about the tradeoff?

Reply:
We discussed this on a call and decided it is fine to represent the relationship of these properties to an external module as long as we own that external module.